### DISCUSSION TITLE
`trpc.error` syntax

### DISCUSSION THREAD

**Discussion by colinhacks** [trpc/trpc#270] [General]
I see the appeal of the `trpc.httpError.forbidden` syntax for common HTTP codes, but I think there should be a fallback function for easily throwing HTTPErrors with any numerical code. Similar to the v0.x syntax:

`trpc.error(code: number, opts; { status?: string; message?: string; originalError?: Error })`

```ts
throw trpc.error(418, { message: "I'm a teapot" })
```

I'm used to thinking about HTTP codes as numbers. It looks like you're trying to require a string representation of the status code (`TCode`) which is understandable. You could use something like this to convert the numerical code into a string code: https://github.com/prettymuchbryce/http-status-codes/blob/master/src/status-codes.ts

Alternatively, just expand `trpc.httpError` to include all possible status codes. It's odd that only certain codes are available.

**Comment by KATT** [comment]
No big opposition here, kinda just mimicking how it is to throw unauth errors etc in graphql servers

**Comment by colinhacks** [comment]
I can submit a PR. Which solution do you prefer? 

1. The `trpc.error()` function (which probably involves refactoring away from `TCode` and all that)
2. Expand httpError using https://github.com/prettymuchbryce/http-status-codes/blob/master/src/status-codes.ts

**Comment by KATT** [comment]
Sooo, actually - -

Thought more about this and remembered why it's not like that. 

I don't think the errors we throw in the router should be tied to http, they aren't now and I use a error.code (taken straight from graphql, I have info in the docs) which is translated to http codes in http middlewares. 

I am planning to do support for other transports later so internal codes should be something different from http.

**Comment by KATT** [comment]
https://trpc.io/docs/error-handling
